src/llvm_ir_generator.cpp

```diff
@@ -163,39 +163,81 @@ void LLVMIRGenerator::Visit(const DeclStmtNode& decl_stmt) {
 
 void LLVMIRGenerator::Visit(const VarDeclNode& decl) {
   auto var_type = builder_helper_.GetLLVMType(*(decl.type));
-  auto addr = builder_.CreateAlloca(var_type);
-  if (decl.init) {
-    decl.init->Accept(*this);
-    auto val = val_recorder.ValOfPrevExpr();
-    builder_.CreateStore(val, addr);
+  if (decl.is_global) {
+    auto global = module_.getOrInsertGlobal(decl.id, var_type);
+    auto global_var = llvm::dyn_cast<llvm::GlobalVariable>(global);
+    llvm::Constant* const_val = nullptr;
+    if (decl.init) {
+      decl.init->Accept(*this);
+      auto val = val_recorder.ValOfPrevExpr();
+      const_val = llvm::dyn_cast<llvm::Constant>(val);
+      assert(const_val);
+    } else {
+      const_val = llvm::ConstantInt::get(builder_.getInt32Ty(), 0, true);
+    }
+    global_var->setInitializer(const_val);
+    id_to_val[decl.id] = global_var;
+  } else {
+    auto addr = builder_.CreateAlloca(var_type);
+    if (decl.init) {
+      decl.init->Accept(*this);
+      auto val = val_recorder.ValOfPrevExpr();
+      builder_.CreateStore(val, addr);
+    }
+    id_to_val[decl.id] = addr;
   }
-  id_to_val[decl.id] = addr;
 }
 
 void LLVMIRGenerator::Visit(const ArrDeclNode& arr_decl) {
-  auto arr_type = builder_helper_.GetLLVMType(*(arr_decl.type));
-  auto base_addr = builder_.CreateAlloca(arr_type);
-  id_to_val[arr_decl.id] = base_addr;
+  auto type = builder_helper_.GetLLVMType(*(arr_decl.type));
+  llvm::AllocaInst* base_addr = nullptr;
+  if (arr_decl.is_global) {
+    auto global_arr = module_.getOrInsertGlobal(arr_decl.id, type);
+    id_to_val[arr_decl.id] = global_arr;
+  } else {
+    auto addr = builder_.CreateAlloca(type);
+    id_to_val[arr_decl.id] = addr;
+    base_addr = addr;
+  }
 
   auto arr_decl_type = dynamic_cast<ArrType*>(arr_decl.type.get());
+  std::vector<llvm::Constant*> arr_elems{};
```

Comment:
Please add a brief comment stating that this vector stores the initial values of the global array.